src/Storages/ObjectStorage/DataLakes/Iceberg/ManifestFileImpl.h

```diff
@@ -46,6 +47,15 @@ struct ManifestFileContentImpl
         Int32 schema_id_);
 
     Int32 schema_id;
+
+    // Size - number of supported partition columns
+    std::vector<PartitionTransform> partition_transforms;
+    std::vector<Int32> partition_source_ids;
+
+    //Size - number of supported partition columns, each column has an entry for each file
+    std::vector<DB::ColumnPtr> partition_columns;
+
+    // Size - number of files
     std::vector<DataFileEntry> data_files;
```

Comment:
Maybe we can calculate the range for each file during manifest file parsing and add a list of ranges inside `DataFileEntry` struct? I think it can simplify the code, because we won't need to cereate prunning mask and check if file should be skipped or not during iteration over `data_files` in `getDataFilesImpl`. WDYT?